File: backend/src/services/ai/recommendation.js
/**
 * src/services/ai/recommendation.js
 *
 * Integration service for the final recommendation synthesis layer.
 */

import aiClient from './client.js';

/**
 * Triggers recommendation synthesis from upstream components.
 *
 * @param {object} payload - Must contain { thesis, review, critique }
 * @returns {Promise<{ status: string, data: object }>}
 */
export const buildRecommendation = async (payload) => {
  const response = await aiClient._request('/api/v1/recommendation/build', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
  return response;
};

File: ai-service/src/api/routes/recommendation.py
# ─────────────────────────────────────────────────────────────────
# src/api/routes/recommendation.py
# ─────────────────────────────────────────────────────────────────
#
# FastAPI API endpoints router for the Investment Recommendation Layer.
# ─────────────────────────────────────────────────────────────────

from fastapi import APIRouter
from src.thesis.models import InvestmentThesis
from src.committee.models import InvestmentCommitteeReview
from src.critique.models import InvestmentCritique
from src.recommendation.builder import RecommendationBuilder
from src.utils.response import send_success, send_error

router = APIRouter()

@router.post("/build")
def build_recommendation(payload: dict):
    """
    Builds the final InvestmentRecommendation deterministically from
    Thesis, CommitteeReview, and Critique models.
    """
    try:
        if "thesis" not in payload or "review" not in payload or "critique" not in payload:
            return send_error(message="Payload must contain 'thesis', 'review', and 'critique' objects.", code=400)
            
        thesis = InvestmentThesis(**payload["thesis"])
        review = InvestmentCommitteeReview(**payload["review"])
        critique = InvestmentCritique(**payload["critique"])
        
        recommendation = RecommendationBuilder.build(thesis, review, critique)
        
        return send_success(data=recommendation.model_dump())
        
    except Exception as e:
        return send_error(message=f"Recommendation building failed: {str(e)}", code=500)
